pkg/controller/node.go

```diff
@@ -273,11 +273,21 @@ func (c *Controller) handleAddNode(key string) error {
 				"node":           node.Name,
 				"address-family": strconv.Itoa(af),
 			}
-			klog.Infof("add policy route for router: %s, match %s, action %s, nexthop %s, extrenalID %v", c.config.ClusterRouter, match, "reroute", ip, externalIDs)
+			klog.Infof("add policy route for router: %s, match %s, action %s, nexthop %s, externalID %v", c.config.ClusterRouter, match, "reroute", ip, externalIDs)
 			if err = c.ovnLegacyClient.AddPolicyRoute(c.config.ClusterRouter, util.NodeRouterPolicyPriority, match, "reroute", ip, externalIDs); err != nil {
 				klog.Errorf("failed to add logical router policy for node %s: %v", node.Name, err)
 				return err
 			}
+
+			if c.config.NodeLocalDnsIP != "" {
+				if err = c.addPolicyRouteForLocalDnsCacheOnNode(portName, ip, node.Name, af); err != nil {
```

Comment:
The NodeLocalDnsIP may change, we should remove legacy routes

Reply:
deletePolicyRouteForLocalDnsCacheOnNode use external id to find the policy route, It will delete the policy routing regardless of whether the NodeLocalDnsIP is changed or not

Reply:
fixed